OpenRA.Mods.Common/Activities/Air/ResupplyAircraft.cs

```diff
@@ -27,22 +28,29 @@ public ResupplyAircraft(Actor self)
 
 		public override Activity Tick(Actor self)
 		{
-			var host = aircraft.GetActorBelow();
-
-			if (host == null)
-				return NextActivity;
-
-			if (aircraft.IsPlane)
-				return ActivityUtils.SequenceActivities(
-					aircraft.GetResupplyActivities(host)
-					.Append(new CallFunc(() => aircraft.UnReserve()))
-					.Append(new WaitFor(() => NextActivity != null || Reservable.IsReserved(host)))
-					.Append(new TakeOff(self))
-					.Append(NextActivity).ToArray());
-
-			// If is helicopter move away as soon as the resupply ends
-			return ActivityUtils.SequenceActivities(
-				aircraft.GetResupplyActivities(host).Append(new TakeOff(self)).Append(NextActivity).ToArray());
+			if (inner == null)
```

Comment:
I don't like that for two reasons:

1. It makes the method harder to read. You'd have the stuff that happens in subsequent ticks at the top, and the thing that runs during the first tick at the bottom.
2. I'd need two return statements.